pymongo/asynchronous/client_bulk.py

```diff
@@ -133,7 +134,9 @@ def add_insert(self, namespace: str, document: _DocumentOut) -> None:
         validate_is_document_type("document", document)
         # Generate ObjectId client side.
         if not (isinstance(document, RawBSONDocument) or "_id" in document):
-            document["_id"] = ObjectId()
+            document = ChainMap(document, {"_id": ObjectId()})
```

Comment:
We still need to add id to the input document here. Can you also add a test for that?